src/transforms/aws_ec2_metadata.rs

```diff
@@ -0,0 +1,596 @@
+type ReadHandle = evmap::ReadHandle<Atom, Bytes, (), RandomState>;
+
+lazy_static::lazy_static! {
+    static ref AMI_ID: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/ami-id");
+    static ref AMI_ID_KEY: Atom = Atom::from("ami-id");
+
+    static ref AVAILABILITY_ZONE: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/placement/availability-zone");
+    static ref AVAILABILITY_ZONE_KEY: Atom = Atom::from("availability-zone");
+
+    static ref INSTANCE_ID: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/instance-id");
+    static ref INSTANCE_ID_KEY: Atom = Atom::from("instance-id");
+
+    static ref LOCAL_HOSTNAME: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/local-hostname");
+    static ref LOCAL_HOSTNAME_KEY: Atom = Atom::from("local-hostname");
+
+    static ref LOCAL_IPV4: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/local-ipv4");
+    static ref LOCAL_IPV4_KEY: Atom = Atom::from("local-ipv4");
+
+    static ref PUBLIC_HOSTNAME: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/public-hostname");
+    static ref PUBLIC_HOSTNAME_KEY: Atom = Atom::from("public-hostname");
+
+    static ref PUBLIC_IPV4: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/public-ipv4");
+    static ref PUBLIC_IPV4_KEY: Atom = Atom::from("public-ipv4");
+
+    static ref REGION: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/region");
+    static ref REGION_KEY: Atom = Atom::from("region");
+
+    static ref SUBNET_ID: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/network/interfaces/macs/mac/subnet-id");
+    static ref SUBNET_ID_KEY: Atom = Atom::from("subnet-id");
+
+    static ref VPC_ID: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/network/interfaces/macs/mac/vpc-id");
+    static ref VPC_ID_KEY: Atom = Atom::from("vpc-id");
+
+    static ref ROLE_NAME: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/iam/security-credentials/");
+    static ref ROLE_NAME_KEY: Atom = Atom::from("role-name");
+
+    static ref MAC: PathAndQuery = PathAndQuery::from_static("/latest/meta-data/mac");
+
+    static ref DYNAMIC_DOCUMENT: PathAndQuery = PathAndQuery::from_static("/latest/dynamic/instance-identity/document");
+
+    static ref DEFAULT_FIELD_WHITELIST: Vec<Atom> = vec![
+        AMI_ID_KEY.clone(),
+        AVAILABILITY_ZONE_KEY.clone(),
+        INSTANCE_ID_KEY.clone(),
+        LOCAL_HOSTNAME_KEY.clone(),
+        LOCAL_IPV4_KEY.clone(),
+        PUBLIC_HOSTNAME_KEY.clone(),
+        PUBLIC_IPV4_KEY.clone(),
+        REGION_KEY.clone(),
+        SUBNET_ID_KEY.clone(),
+        VPC_ID_KEY.clone(),
+        ROLE_NAME_KEY.clone(),
+    ];
+
+    static ref API_TOKEN: PathAndQuery = PathAndQuery::from_static("/latest/api/token");
+    static ref TOKEN_HEADER: Bytes = Bytes::from("X-aws-ec2-metadata-token");
+    static ref HOST: Uri = Uri::from_static("http://169.254.169.254");
```

Comment:
Do they not care about https for this?

Reply:
Yeah, none of the curl examples or example libraries used https. I assume its all instance local or rack local.